src/analysis/normed_space/add_torsor.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2020 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Joseph Myers.
+-/
+import algebra.add_torsor
+import analysis.normed_space.basic
+
+noncomputable theory
+
+/-!
+# Torsors of additive normed group actions.
+
+This file defines torsors of additive normed group actions, with a
+metric space structure.  The motivating case is Euclidean affine
+spaces.
+
+-/
+
+section prio
+set_option default_priority 100 -- see Note [default priority]
+/-- A `normed_add_torsor V P` is a torsor of an additive normed group
+action by a `normed_group V` on points `P`. We bundle the metric space
+structure and require the distance to be the same as results from the
+norm (which in fact implies the distance yields a metric space, but
+bundling just the distance and using an instance for the metric space
+results in type class problems). -/
+class normed_add_torsor (V : Type*) (P : Type*) [normed_group V] [metric_space P]
+  extends add_torsor V P :=
+(norm_dist' : ∀ (x y : P), dist x y = ∥(x -ᵥ y : V)∥)
+end prio
+
+/-- The distance equals the norm of subtracting two points. This lemma
+is needed to make V an explicit rather than implicit argument. -/
+lemma norm_dist (V : Type*) {P : Type*} [normed_group V] [metric_space P] [normed_add_torsor V P]
```

Comment:
In `normed_group` it's called `dist_eq_norm`. This could be `dist_eq_norm'` or `torsor_dist_eq_norm`.

Reply:
Do you assume that users should apply this lemma then deal with `norm`, or that users should be able to deal with `dist` in the original space without going to `norm`? In the latter case it makes sense to add `simp` lemmas like `dist (v +ᵥ x) (v +ᵥ y) = dist x y`.

Reply:
Renamed to `torsor_dist_eq_norm`. I'm expecting users to be applying this lemma then dealing with `norm`, or at least that's what I'm doing in Euclidean affine spaces, but it's entirely possible people will find other ways of using this code.

Reply:
Then probably this should be a `simp` lemma.